packages/button/src/Button.tsx

```diff
@@ -68,11 +69,14 @@ const Button: ExtendableBox<ButtonProps & { ref?: React.Ref<any> }, 'button'> =
       className: cx(buttonClassName, className),
       ref: forwardRef,
       // Provide a default value for the as prop
-      // If consumping application passes a value for as, it will override the default set here
-      as: (isAnchor ? 'a' : 'button') as keyof JSX.IntrinsicElements,
+      // If consuming application passes a value for as, it will override the default set here
+      as: as
+        ? as
+        : ((isAnchor ? 'a' : 'button') as keyof JSX.IntrinsicElements),
       // only add a disabled prop if not an anchor
       ...(typeof rest.href !== 'string' && { disabled }),
       'aria-disabled': disabled,
+      onClick: !disabled ? onClick : undefined,
```

Comment:
If the element type is not `button`, the `onClick` prop won't automatically be ignored when `disabled` is true.